spec/factories/users.rb

```diff
@@ -22,12 +22,14 @@
     signup_cta_variant           { "navbar_basic" }
     email_digest_periodic        { false }
 
-    after(:create) do |user|
-      create(:identity, user_id: user.id)
-    end
+    trait :with_identity do
+      transient { identities { %i[github twitter] } }
 
-    trait :two_identities do
-      after(:create) { |user| create(:identity, user_id: user.id, provider: "twitter") }
+      after(:create) do |user, options|
+        Array.wrap(options.identities).each do |provider|
```

Comment:
Tbh, I'd probably remove the `Array.wrap` here and have users of the factory use arrays explicitly. It's a bit confusing to read when it's called with a plain string and an array in the same spec,

Reply:
I'm glad you pointed this out because I was a bit hesitant about this. Updated!